base/inference.jl

```diff
@@ -52,20 +52,39 @@ const NF = NotFound()
 typealias LineNum Int
 typealias VarTable Array{Any,1}
 
+# The type of a variable load is either a value or an UndefVarError
 type VarState
     typ
     undef::Bool
     VarState(typ::ANY, undef::Bool) = new(typ, undef)
 end
 
-immutable Const
+# The type of a value might be constant
+type Const
```

Comment:
no, we should never modify it in-place. But when we can start stack-allocating immutables, we want to make sure this still gets heap-allocated.